pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1283,12 +1283,7 @@ cdef class RelativeDeltaOffset(BaseOffset):
             else:
                 td_nano = Timedelta(0)
 
-            if self.n > 0:
-                for i in range(self.n):
-                    other = other + self._offset + td_nano
-            else:
-                for i in range(-self.n):
-                    other = other - self._offset - td_nano
+            other = other + ((self._offset + td_nano) * self.n)
```

Comment:
What's the motivation here?

Reply:
The former version of this code adds the number `n` of multiplied offsets one by one in a loop.  This means that it is possible for cutoffs to occur that will change the final value.  For example, if I were to take a date of January 31st, and then add 2 months to this date by performing `date + (offset * 2)` where `date` is the original date and `offset` is `DateOffset(months=1)`, I would get different results in these 2 versions.

In the old version, I would obtain March 28th or 29th based on the year.  This is because of the loop, which causes 1 month to be added twice.  After the first addition, the offset would find itself at February 31st, but since that doesn't exist, it would move itself to the last day of February.  Then, during the second addition, it would simply add another month, ending up with March 28th or 29th.

This is not consistent with how this functionality works when the addition is done to an array or series of dates, which return March 31st in this situation.  The new version would give back March 31st, since it would take the sum of all the multiplied offsets and adds them all at once.  This prevents the date being changed due to things that the operation passes over.

Reply:
Thx very much for the explanation